src/lib/features/feature-toggle/deep-diff.ts

```diff
@@ -0,0 +1,66 @@
+interface Difference {
+    index: (string | number)[];
+    reason: string;
+    valueA: any;
+    valueB: any;
+}
+
+export function deepDiff(arr1: any[], arr2: any[]): Difference[] | null {
```

Comment:
is there relationship between the arr1 and arr2 types? 
`deepDiff<T>(arr1: T[], arr2: T[]): Difference[] | null`

Reply:
In this case they are different types.